src/Symfony/Component/Messenger/Tests/Transport/AmqpExt/AmqpTransportFactoryTest.php

```diff
@@ -36,8 +36,8 @@ public function testItCreatesTheTransport()
         $factory = new AmqpTransportFactory();
         $serializer = $this->createMock(SerializerInterface::class);
 
-        $expectedTransport = new AmqpTransport(Connection::fromDsn('amqp://localhost', ['foo' => 'bar']), $serializer);
+        $expectedTransport = new AmqpTransport(Connection::fromDsn('amqp://localhost', ['host' => 'localhost']), $serializer);
```

Comment:
Symptom of a BC break, isn't it? Should we trigger a deprecation first and start throwing in 6.0?

Reply:
/cc @weaverryan  WDYT?

we had a chat on slack about that, and we've concluded that this is not a BC break. Because these options had no effect before.

but maybe we were wrong...

Reply:
Well, it was silently ignored, now it would throw an exception, breaking workers at runtime. Seems worth a BC layer IMHO

Reply:
Hmm, that's a good point about workers. The exception would be obvious, but as workers are an automated thing, it could cause a surprising failure on deploy. I guess we should add a BC layer here.

Reply:
OK, i've removed the exceptions in favor of deprecations.  
It's the first time i deprecate something in symfony. Following the docs, i've added an entry in the CHANGELOG.md